### PR TITLE
Fix compilation errors for FreeBSD 12.0

### PR DESCRIPTION
**Pull request by vlvkobal**
Fixes https://github.com/firehol/netdata/pull/1321#issuecomment-333936592

### CODE REVIEW

File: src/freebsd_sysctl.c [hunk 1]
```diff
@@ -27,6 +27,59 @@
 
 int system_pagesize = PAGE_SIZE;
 int number_of_cpus = 1;
+#if __FreeBSD_version >= 1200029
```
Comment (ktsaou) [inline src/freebsd_sysctl.c]: Is this a typo?
They removed the structure after this version?

File: src/freebsd_sysctl.c [hunk 2]
```diff
@@ -856,7 +909,11 @@ int do_system_ram(int update_every, usec_t dt) {
     (void)dt;
     static int mib_active_count[4] = {0, 0, 0, 0}, mib_inactive_count[4] = {0, 0, 0, 0}, mib_wire_count[4] = {0, 0, 0, 0},
                mib_cache_count[4] = {0, 0, 0, 0}, mib_vfs_bufspace[2] = {0, 0}, mib_free_count[4] = {0, 0, 0, 0};
+#if __FreeBSD_version < 1200029
```
Comment (ktsaou) [inline src/freebsd_sysctl.c]: You can avoid these `#ifdef` statements with a `typedef`. I suggest to `typedef` the right one at the struct definition, to keep the code clean...

File: src/freebsd_sysctl.c (hunk 1 above)
Comment (vlvkobal) [reply src/freebsd_sysctl.c]: They changed and hid it. They also defined `__vmmeter` as a workaround in `vmstat` utility.

**Comment by vlvkobal** [comment]
Thank you for the hint. I've changed #ifdefs to typedefs.

**Comment by ktsaou** [comment]
merged! thanks!

**Pull request merged**